### ISSUE TITLE
Back button improvements

### ISSUE THREAD

**Issue by Macroz** [CSCfi/rems#1339]
Go through all the back buttons and update them

- a back button can be removed if it does the same thing as the browser back button
- a back button that should take you to higher level in navigation (from detail to list) can be removed if there is a navigation link close by. It should never be named back button but rather where it takes (e.g. Forms)
- "back" button should be on the left side of the actions area?
- consider adding a button to navigate back to the top on a long page

**Comment by okahilak** [comment]
I opted to not do anything about this issue now. Some arguments for that below:

- a back button can be removed if it does the same thing as the browser back button

Back buttons are currently shown the the user when viewing a created catalogue item, resource, form, workflow, or license in the administration menu.

One can end up in that view in one of two ways: by either viewing a previously created item or creating a new item. In the former case, "Back" button does the same as clicking the Back button in browser. However, in the latter case, the browser's Back button leads to the form with which the item is created.

To do this task, the Back button would need to be removed when viewing a previously created item but visible when a new item has just been created. That would probably feel too inconsistent for the user, so I'm leaning towards leaving the Back button as is.

- a back button that should take you to higher level in navigation (from detail to list) can be removed if there is a navigation link close by. It should never be named back button but rather where it takes (e.g. Forms)

To me, having a button that reads "Forms" is more difficult to understand than simple "Back", which immediately gives the impression that I'm taken somewhere where I was previously, which is easy to associate with the list that I was reading just before (if viewing a previously created item) or a few steps back (if I have just created a new item).

The navigation link is sometimes close by to the "Back" button, e.g., in the case of viewing a license, which often does not take much space from the page, and sometimes it is farther away, e.g., in the case of resource with a long list of licenses. For consistency, I'd like to keep "Back" button there regardless.

- "back" button should be on the left side of the actions area?

I think the user quickly gets accustomed to many buttons being located in the right side of the window, mostly because in the tables, the buttons naturally fall on the right side of a table row. "Back" button would be the only exception, so I think that would then need more effort from the user to locate compared to it being in the "usual" place.

Also, the "Back" button is often accompanied with, e.g., "Edit" button, as well as other action buttons, which are typically located on the right side, and having all the buttons close to each other has the benefit of the user not having to visually scan the page to find all the possible actions to take.

- consider adding a button to navigate back to the top on a long page

This could be done, but I personally don't feel the need for that: using the scrollbar to navigate to the top of the page would probably be enough for me. But some others might disagree -- I'm open for discussion about this.

Overall, going through all locations where Back button is used and testing a bit, I'm relatively happy with how it works at the moment. Some of the changes proposed are such that even if they improve the user experience a bit, I think they wouldn't play a big part in any usability problems that the system might currently have.
